### PR TITLE
Migrate project to Jetpack Compose

### PR DESCRIPTION
**Pull request by haroldadmin**
This PR migrates WhatTheStack to use [Jetpack Compose](https://developer.android.com/jetpack/compose), Android's new declarative UI toolkit.

- Migrate the library module to Compose
- Migrate the sample app to Compose
- Use accompanist SystemUI Controller and Insets module to handle immersive mode

I've maintained feature parity with the existing stable version of the library, but will
release this as a new major release after it's been tested through a few pre-release builds.

### CODE REVIEW

File: what-the-stack/build.gradle [hunk 1]
```diff
@@ -36,7 +36,11 @@ android {
     }
 
     buildFeatures {
-        viewBinding true
+        compose true
+    }
+
+    composeOptions {
+        kotlinCompilerExtensionVersion "1.0.5"
```
Comment (msfjarvis) [inline what-the-stack/build.gradle]: After making the change in libs.version.toml

```suggestion
        kotlinCompilerExtensionVersion libs.versions.compose.get()
```

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -17,9 +17,18 @@ espressoCore = "3.2.0"
 mockk = "1.9.3"
 robolectric = "4.3.1"
 startup = "1.0.0"
+composeActivity = "1.3.1"
+composeMaterial = "1.0.5"
+composeTooling = "1.0.5"
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]: These will always be the same so renaming this to `compose` and re-using should be fine.

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/ui/components/IconOutlinedButton.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+
+@Composable
+internal fun OutlinedIconButton(
```
Comment (msfjarvis) [inline what-the-stack/src/main/java/com/haroldadmin/whatthestack/ui/components/IconOutlinedButton.kt]: The file name is `IconOutlinedButton`, you should probably rename it to match the function.

File: what-the-stack/src/main/java/com/haroldadmin/whatthestack/ui/components/ScrollableText.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+
+@Composable
+internal fun ScrollableText(
```
Comment (msfjarvis) [inline what-the-stack/src/main/java/com/haroldadmin/whatthestack/ui/components/ScrollableText.kt]: This is only used once so it doesn't make a lot of sense to separate it out, unless you have plans to make use of it in the future.

**Review by msfjarvis — COMMENTED** [review]
LGTM, great work! I've left some minor cleanup related comments but none are a blocker.

File: what-the-stack/build.gradle (hunk 1 above)
Comment (haroldadmin) [reply what-the-stack/build.gradle]: TIL. Didn't know the DSL for versions catalog supported such operations. Fixed now.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (haroldadmin) [reply gradle/libs.versions.toml]: Thanks, fixed now.

**Pull request merged**